src/UI/Art/SpriteArtist/PlayerSetupArtist.java

```diff
@@ -0,0 +1,282 @@
+      teamLabel = new SpriteUIUtils.ImageFrame(65, 12, 28, 10, MENUHIGHLIGHTCOLOR, MENUBGCOLOR, false, SpriteLibrary.getTextAsImage("TEAM"));
+      aiLabel = new SpriteUIUtils.ImageFrame(94, 12, 54, 10, MENUHIGHLIGHTCOLOR, MENUBGCOLOR, false, SpriteLibrary.getTextAsImage("CONTROL"));
+      teamLabel.render(g);
+      aiLabel.render(g);
+
+      update(info);
+    }
+
+    public BufferedImage update(PlayerSetupInfo info)
+    {
+      // Pull out the info we need.
+      // "Commander-Color Faction"
+      StringBuffer coStrBuf = new StringBuffer(info.getCurrentCO().name);
+      coStrBuf.append("-").append(UIUtils.getPaletteName(info.getCurrentColor())).append(" ").append(info.getCurrentFaction().name);
+
+      boolean cmdrChanged = !info.getCurrentCO().name.equals(commanderName);
+      boolean colorChanged = !UIUtils.getPaletteName(info.getCurrentColor()).equals(colorName);
+      boolean factionChanged = !info.getCurrentFaction().name.equals(factionName);
+      boolean teamChanged = teamNumber != info.getCurrentTeam();
+      boolean aiChanged = !info.getCurrentAI().getName().equals(aiName);
+
+      Graphics g = myImage.getGraphics();
+      if( cmdrChanged || factionChanged )
+      {
+        commanderName = info.getCurrentCO().name;
+        colorName = UIUtils.getPaletteName(info.getCurrentColor());
+        factionName = info.getCurrentFaction().name;
+        descriptionPane = new SpriteUIUtils.ImageFrame(1, 1, PANEL_WIDTH - 2, 10, MENUHIGHLIGHTCOLOR, MENUBGCOLOR, false, SpriteLibrary.getTextAsImage(coStrBuf.toString()));
+        descriptionPane.render(g);
+      }
+      if( cmdrChanged || colorChanged )
+      {
+        commanderPane = new SpriteUIUtils.ImageFrame(1, 12, portraitPx + 2, portraitPx + 2, info.getCurrentColor(),
+            info.getCurrentColor(), true, SpriteLibrary.getCommanderSprites( info.getCurrentCO().name ).head);
+        commanderPane.render(g);
+      }
+      if( factionChanged || colorChanged )
+      {
+        UnitSpriteSet inf = SpriteLibrary.getMapUnitSpriteSet(UnitModel.UnitEnum.INFANTRY, info.getCurrentFaction(), info.getCurrentColor());
+        BufferedImage infSprite = inf.sprites[inf.ACTION_IDLE].getFrame(0);
+        unitPane = new SpriteUIUtils.ImageFrame(portraitPx + 4, 12, 28, portraitPx + 2, MENUBGCOLOR, MENUHIGHLIGHTCOLOR, true, infSprite);
+        unitPane.render(g);
+      }
+      if( teamChanged )
+      {
+        teamNumber = info.getCurrentTeam();
+        teamPane = new SpriteUIUtils.ImageFrame(65, 23, 28, 23, MENUBGCOLOR, MENUHIGHLIGHTCOLOR, true,
+            SpriteLibrary.getMapUnitHPSprites().getFrame(info.getCurrentTeam()));
+        teamPane.render(g);
+      }
+      if( aiChanged )
+      {
+        aiName = info.getCurrentAI().getName();
+        aiPane = new SpriteUIUtils.ImageFrame(94, 23, 54, 23, MENUBGCOLOR, MENUHIGHLIGHTCOLOR, true, SpriteLibrary.getTextAsImage(info.getCurrentAI().getName()));
+        aiPane.render(g);
+      }
```

Comment:
Just a note, there are an awful lot of magic numbers here.

Reply:
Yeah. I could defend that by saying it's a rigidly-defined format that isn't expected to change, and isn't any different than defining a spec for a binary data format, but really I was just trying to get a working prototype quickly so I could iterate from there.
I may go back and un-magicify things, unless I conveniently forget.